bindings.c

```diff
@@ -4045,9 +4075,48 @@ static int read_cpuacct_usage_all(char *cg, char *cpuset, struct cpuacct_usage *
 	if (!cpu_usage)
 		return -ENOMEM;
 
+	memset(cpu_usage, 0, sizeof(struct cpuacct_usage) * cpucount);
 	if (!cgfs_get_value("cpuacct", cg, "cpuacct.usage_all", &usage_str)) {
-		rv = -1;
-		goto err;
+		lxcfs_v("failed to read cpuacct.usage_all%s\n", "");
+
+		// read cpuacct.usage_percpu instead
+		lxcfs_v("reading cpuacct.usage_percpu instead%s\n", "");
+		if (!cgfs_get_value("cpuacct", cg, "cpuacct.usage_percpu", &usage_str)) {
+			rv = -1;
+			goto err;
+		}
+
+		// convert cpuacct.usage_percpu into cpuacct.usage_all
+		lxcfs_v("converting cpuacct.usage_percpu into cpuacct.usage_all%s\n", "");
+		stringbuilder *sb = sb_create();
+		if (NULL == sb) {
+			rv = -1;
+			goto err;
+		}
+		if (SB_FAILURE == sb_append(sb, "cpu user system\n")) {
+			sb_free(sb);
+			rv = -1;
+			goto err;
+		}
+		int i = 0;
+		while (sscanf(usage_str + read_pos, "%lu %n", &cg_user, &read_cnt) > 0) {
+			lxcfs_debug("i: %d, cg_user: %lu, read_pos: %d, read_cnt: %d\n", i, cg_user, read_pos, read_cnt);
+			if (SB_FAILURE == sb_appendf(sb, "%d %lu %lu\n", i, cg_user, 0)) {
+				sb_free(sb);
+				rv = -1;
+				goto err;
+			}
+			i++;
+			read_pos += read_cnt;
+		}
+
+		free(usage_str);
+		usage_str = sb_concat(sb);
+		sb_free(sb);
+		read_pos=0;
+		read_cnt=0;
```

Comment:
The next access to `read_pos` includes a read (`+=`), which makes me really uncomfortable, despite the only thing happening before being an initialization to `0`, because that can now easily get lost in patch contexts.
I'd rather have `read_pos/cnt` shadowed in this block, and the initialization of the "outer" `read_pos` moved down.